### ISSUE TITLE
Error when creating a Drive

### ISSUE THREAD

**Issue by jannishuebl** [hanneshayashi/terraform-provider-gdrive#11]
When i want to create a shared drive, the drive is created but terraform gives me:

```
Error: XXXXX: googleapi: Error 404: Shared drive not found: XXXXX, notFound
````

I think there could be some timeing issue:
Do you think it could help to increase this sleep time?

https://github.com/hanneshayashi/terraform-provider-gdrive/blob/65cb3acc567b7d6e1c0850ae29993861f36b3078/provider/resource_drive.go#L149

**Comment by hanneshayashi** [comment]
Hi and thanks for the bug report!
The API is unfortunately a little inconsistent sometimes. There is actually code in GSM that is supposed to make the function only return the Drive object after it can be queried by the API by checking the permissions on it:
https://github.com/hanneshayashi/gsm/blob/main/gsmdrive/drives.go#L51-L61
This seems to work in most cases. I also tried querying it again immediately after creation, but this will just work most of the time but subsequent calls will still fail. It just seems that the backend takes a while to replicate the information about the Drive...

Since the API is really inconsistent, it is also hard to test this behaviour.

The sleep timer should just be a failsafe, but it looks like it may not be enough. I could definitely try to increase it to 10 seconds, although I would prefer a more elegant and permanent solution.

I will take a look at this for the next release. As an immediate workaround you could try something with https://registry.terraform.io/providers/hashicorp/time/latest/docs/resources/sleep.

**Comment by hanneshayashi** [comment]
@jannishuebl I was able to reproduce the issue and it still boils down to the API not being strongly consistent. After the creation of a Drive, the API immediately returns the object, but subsequent calls that reference or use the Drive may result in 404 errors for a while. Even if one or two requests succeed, the next one may still return 404.

The workaround for now is to have a longer wait time to give the backend time to replicate. I just released version v0.9.1 of the provider that introduces a new optional parameter called `wait_after_create` with a default value of 60 that allows you to configure the time the provider waits after the initial creation of the Drive object. Please test it out and let me know if this fixes the issue or re-open this issue if it still doesn't work.

As a side-note: In hindsight, it might have been better to create a separate resource for Drive restriction, as these cannot be set during the creation and need an update operation after the Drive has been created. I may change this in a v1.0 of the provider, as it would be a breaking change.
